sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java

```diff
@@ -89,17 +90,94 @@ public PollerFlux(Duration defaultPollInterval,
+     * @param cancelOperation a {@link Function} that represents the operation to cancel the long-running operation
+     *                        if service supports cancellation, this parameter is required and if service does not
+     *                        support cancellation then the implementer should return Mono.error with an error message
+     *                        indicating absence of cancellation support, the operation will be called with current
+     *                        {@link PollingContext}.
+     * @param fetchResultOperation a {@link Function} that represents the  operation to retrieve final result of
+     *                             the long-running operation if service support it, this parameter is required and
+     *                             operation will be called current {@link PollingContext}, if service does not have an
+     *                             api to fetch final result and if final result is same as final poll response value
+     *                             then implementer can choose to simply return value from provided final poll response.
+     *
+     * @param <T> The type of poll response value.
+     * @param <U> The type of the final result of long-running operation.
+     * @return PollerFlux
+     */
+    public static <T, U> PollerFlux<T, U>
+        create(Duration defaultPollInterval,
+               Function<PollingContext<T>, Mono<PollResponse<T>>> activationOperation,
+               Function<PollingContext<T>, Mono<PollResponse<T>>> pollOperation,
+               BiFunction<PollingContext<T>, PollResponse<T>, Mono<T>> cancelOperation,
+               Function<PollingContext<T>, Mono<U>> fetchResultOperation) {
+        return new PollerFlux<>(defaultPollInterval,
+            activationOperation,
+            pollOperation,
+            cancelOperation,
+            fetchResultOperation,
+            true);
+    }
+
+    private PollerFlux(Duration defaultPollInterval,
+                       Function<PollingContext<T>, Mono<PollResponse<T>>> activationOperation,
+                       Function<PollingContext<T>, Mono<PollResponse<T>>> pollOperation,
+                       BiFunction<PollingContext<T>, PollResponse<T>, Mono<T>> cancelOperation,
+                       Function<PollingContext<T>, Mono<U>> fetchResultOperation,
+                       boolean ignored) {
+        Objects.requireNonNull(defaultPollInterval, "'defaultPollInterval' cannot be null.");
+        if (defaultPollInterval.compareTo(Duration.ZERO) <= 0) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException(
+                "Negative or zero value for 'defaultPollInterval' is not allowed."));
+        }
+        this.defaultPollInterval = defaultPollInterval;
+        this.activationOperation = null;
+        this.activationOperationEx = Objects.requireNonNull(activationOperation,
```

Comment:
nit: rename this to `activationOperationWithResponse` to make it clear that this is different from `activationOperation` which doesn't have the PollResponse.

Reply:
I'm trying to get rid of storing activationOperation in class level, just updated the code to move all activation logic to its own inner class.